python/mpt/src/mpt/utils.cairo

```diff
@@ -123,3 +124,50 @@ func deserialize_to_internal_node{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}
     );
     return result;
 }
+
+// @notice Checks if a branch node is valid.
+// A branch node is valid if it has at least two non-null subnodes.
+// @dev Raises an error if the branch node is invalid.
+// @param node The branch node to check.
+func check_branch_node(node: BranchNode) {
+    alloc_locals;
+
+    local first_non_null_branch;
+    local seconds_non_null_branch;
+    let subnodes_ptr = cast(node.value.subnodes.value, felt*);
+    %{
+        node = serialize(ids.node);
+        non_null_branches = [idx for idx, subnode in enumerate(node.subnodes) if (subnode is not None and subnode != b"")]
```

Comment:
here since we check difference wrt `None` and empty bytes, writing directly memory accesses is a bit ugly cc @enitrat do you agree?

Reply:
i prefer the old way. no use of `serialize`